### ISSUE TITLE
API_DOC endpoint is mandatory after upgrade to ApiPlatform 4.0

### ISSUE THREAD

**Issue by jkabat** [api-platform/core#6848]
**API Platform version(s) affected**: 4.0.10

**Description**  
After upgrade my JSON-LD API does not work when I have setting API_DOC: FALSE (I do not want to expose it for production)

**Additional Context**  
I'm getting RouteNotFoundException

> Unable to generate a URL for the named route "api_doc" as such route does not exist.

Am I missing something?

**Comment by petski** [comment]
I'm getting this whenever I throw an exception. The `RouteNotFoundException` comes from [`vendor/api-platform/hydra/State/HydraLinkProcessor.php:48`](https://github.com/api-platform/hydra/blob/v4.0.11/State/HydraLinkProcessor.php#L48) in my case.

My setup:

- I'm running `v4.0.11`
- `.env.dev` with `DOCS_ENABLE=false`
- `config/packages/api_platform.yaml` has:
    ```
    ...snip...
    enable_docs: "%env(bool:DOCS_ENABLE)%"
    enable_entrypoint: "%env(bool:DOCS_ENABLE)%"
    ...snip...
    ```
- Env resolving works just fine:
    ```
    $ bin/console debug:config --resolve-env api_platform | egrep 'enable_(docs|entrypoint)'
        enable_docs: false
        enable_entrypoint: false
    ```

**Now the curious part (!!):** If I bypass the `%env`-way, and set both values to `false` directly, I do not hit this issue (!).

@jkabat Can you confirm you have this too?

Note to self: https://github.com/api-platform/core/issues?q=is%3Aissue+api_doc+is%3Aclosed has a couple of related issues

**Comment by soyuka** [comment]
The `HydraLinkProcessor` gets removed at: 

https://github.com/api-platform/core/blob/bd0e92936f82d3cd4563cd45ebf1f73fd1db9f01/src/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php#L540-L543

Do you still experience this? What's the value of the `api_platform.enable_docs` parameter?

**Comment by petski** [comment]
The value of `$config['enable_docs']` on line 539 of `vendor/api-platform/symfony/Bundle/DependencyInjection/ApiPlatformExtension.php` is `env_6a1104280fc2a530_bool_DOCS_ENABLE_04563426aed8d72ea9b01c28f8183a50` (which will evaluate to `true`).

As mentioned in my previous comment, it's true value in the config is `"%env(bool:DOCS_ENABLE)%"`

HTH

**Comment by soyuka** [comment]
Not sure how to fix this it's like your env variable didn't got updated?

**Comment by jkabat** [comment]
I used some temporary solution for now (found here in issues). When I get some time, I will do more research regarding your findings.

**Comment by petski** [comment]
Meanwhile, can you please share your temporary solution? Please bump this issue once you've found the root cause.

**Comment by petski** [comment]
https://github.com/api-platform/core/pull/7062 seems related

**Comment by soyuka** [comment]
issue from 2024 and then pr from 2025 ? Does #7062 fixes the issue? please open a new issue with a reproducer

**Comment by petski** [comment]
Sorry for the confusion, @soyuka. After this issue (#6848) got closed, I ended up decorating `api_platform.route_loader` and inject a `api_doc` route if not present as a workaround. Yesterday I bumped to API Platform `v4.1.7` and the workaround I had in place wasn't needed anymore. Iterating the changelog, I suspected #7062 to be the fix for this issue (#6848), hence the "#7062 seems related" remark. I meant to say "#7062 seems a possible fix for this issue". Mea culpa
